### PR TITLE
Add radio buttons & checkboxes

### PR DESCRIPTION
**Pull request by maya**
This adds radio buttons :radio: and checkboxes :heavy_check_mark: to the library. 

This addresses #210.

This is what it looks like:

screen shot 2015-07-23 at 1 33 34 pm

### CODE REVIEW

File: assets/_scss/elements/_inputs.scss [hunk 1]
```diff
@@ -57,4 +57,94 @@ label {
...
+    height: 1.8rem;
+    margin-top: 0;
+    margin-right: 0.5em;
+    margin-bottom: 0;
+    vertical-align: -4px;
+    padding-top: 4px; //compensate for FA vertical-align
+    width: 1.8rem;
+  }
+
+  input[type="radio"] + span {
```
Comment (msecret) [inline assets/_scss/elements/_inputs.scss]: Will this or does this always have to be a `<span>`? Is there a chance that the user might want to make it another element besides a `<span>`?

File: assets/_scss/elements/_inputs.scss [hunk 2]
```diff
@@ -57,4 +57,94 @@ label {
...
+label.radio {
+  display: inline-block;
+  font-weight: 400;
+  margin-bottom: 0.5em;
+
+  input[type="checkbox"] + span {
+    box-shadow: 0 0 0 1px $color-grey-600;
+    border-radius: $usa-border-radius;
+    display: inline-block;
+    height: 1.8rem;
```
Comment (msecret) [inline assets/_scss/elements/_inputs.scss]: Can you do 

```
margin {
  top:
  ...
}

File: assets/_scss/elements/_inputs.scss (hunk 1 above)
Comment (maya) [inline assets/_scss/elements/_inputs.scss]: Good point. @meiqimichelle wrote it this way so I wonder if using something else, like a class, would be more robust.

File: assets/_scss/elements/_inputs.scss [hunk 3]
```diff
@@ -57,4 +57,94 @@ label {
...
+    content: x;
+    position: relative;
+    left: 2px;
+    top: -5px;
+  }
+}
+
+label.radio :checked + span {
+  background-color: $color-primary;
+  border: 2px solid $color-white !important;
```
Comment (msecret) [inline assets/_scss/elements/_inputs.scss]: I'm not sure the `!important` rule is good here.

File: assets/_scss/elements/_inputs.scss [hunk 4]
```diff
@@ -57,4 +57,94 @@ label {
...
+  @include appearance(none);
+  display: inline-block;
+  height: 1.8rem;
+  width: 1.8rem;
+  vertical-align: -4px;
+
+  // to hide checkbox/radio from visual but 
+  // still read to screenreaders
+  border: 0;
+  clip: rect(0 0 0 0);
```
Comment (msecret) [inline assets/_scss/elements/_inputs.scss]: Will this work on all our supported browsers?

File: assets/_scss/elements/_inputs.scss (hunk 2 above)
Comment (maya) [inline assets/_scss/elements/_inputs.scss]: Done in b41a36b.

fyi, it's 

```
margin: {
  top:
  ...
}
```

Need the colon after margin.

File: assets/_scss/elements/_inputs.scss (hunk 1 above)
Comment (juliaelman) [inline assets/_scss/elements/_inputs.scss]: +1 on making it a class and not locking into `<span>`

**Comment by maya** [comment]
:+1: from @nickbristow on 508

**Comment by arvalarva** [comment]
18f's writing style guide is using sentence case. You're using title case in this component. Can you explain why? We go back and forth with this issue A LOT here at the USPTO.

**Comment by maya** [comment]
@arvalarva yes, we will be using sentence case. Fixed in 3da21a7. Will be updating throughout the site too. Updated the screenshot as well. Thanks for your feedback!

File: assets/_scss/elements/_inputs.scss (hunk 4 above)
Comment (maya) [inline assets/_scss/elements/_inputs.scss]: This will: 2844baa.

File: assets/_scss/elements/_inputs.scss [hunk 5]
```diff
@@ -57,4 +57,88 @@ label {
...
+      outline: none;
+    }
+  }
+}
+
+.usa-fieldset-inputs {
+  label {
+    margin-top: 0;
+  }
 }
```
Comment (juliaelman) [inline assets/_scss/elements/_inputs.scss]: New line at end of file.

File: assets/_scss/elements/_typography.scss [hunk 6]
```diff
@@ -116,7 +116,7 @@ dfn {
 }
 
 .usa-sans,
-.usa-sans p {
+.usa-sans * {
```
Comment (juliaelman) [inline assets/_scss/elements/_typography.scss]: Why are we using a wildcard here? Not sure this is best practice and would be hard for users to un-tangle.

**Comment by juliaelman** [comment]
Closing this PR, as it looks as if #227 is an updated version to an updated fix.

**Pull request closed without merging**